cli/cmd/check.go

```diff
@@ -266,39 +269,45 @@ func runExtensionChecks(cmd *cobra.Command, wout io.Writer, werr io.Writer, opts
 			mcCheckCmd := mcCmd.NewCmdCheck()
 			err = executeExtensionCheck(cmd, "multicluster", mcCheckCmd.Flags())
 		default:
-			// Since we don't have checks for extensions we don't support
-			// create a healthchecker that checks if the namespace exists
-			categoryID := healthcheck.CategoryID(ns.Labels[k8s.LinkerdExtensionLabel])
-			checkers := []healthcheck.Checker{}
-			checkers = append(checkers,
-				*healthcheck.NewChecker(fmt.Sprintf("%s extension Namespace exists", categoryID)).
-					WithHintAnchor(fmt.Sprintf("%s-ns-exists", categoryID)).
-					Fatal().
-					WithCheck(func(ctx context.Context) error {
-						_, err := kubeAPI.GetNamespaceWithExtensionLabel(ctx, string(categoryID))
-						if err != nil {
-							return err
-						}
-						return nil
-					}))
-			hc := healthcheck.NewHealthChecker([]healthcheck.CategoryID{categoryID},
-				&healthcheck.Options{
-					ControlPlaneNamespace: controlPlaneNamespace,
-					CNINamespace:          cniNamespace,
-					DataPlaneNamespace:    opts.namespace,
-					KubeConfig:            kubeconfigPath,
-					KubeContext:           kubeContext,
-					Impersonate:           impersonate,
-					ImpersonateGroup:      impersonateGroup,
-					APIAddr:               apiAddr,
-					VersionOverride:       opts.versionOverride,
-					RetryDeadline:         time.Now().Add(opts.wait),
-					CNIEnabled:            opts.cniEnabled,
-				})
-			hc.AppendCategories(*healthcheck.NewCategory(categoryID, checkers, true))
-			success := healthcheck.RunChecks(wout, werr, hc, opts.output)
-			if !success {
-				os.Exit(1)
+			extensionCmd := fmt.Sprintf("linkerd-%s", extension)
+			var path string
+			path, err = exec.LookPath(extensionCmd)
+			if err != nil {
+				if opts.output == healthcheck.JSONOutput {
+					result := map[string]interface{}{
+						"success": false,
+						"categories": []map[string]interface{}{
+							map[string]interface{}{
+								"categoryName": extensionCmd,
+								"checks": []map[string]string{
+									map[string]string{
+										"description": fmt.Sprintf("Linkerd extension command %s exists", extensionCmd),
+										"result":      "failure",
+									},
+								},
+							},
+						},
+					}
+					var out []byte
+					out, err = json.MarshalIndent(result, "", "  ")
+					if err != nil {
+						return err
+					}
+					fmt.Fprintln(wout, string(out))
+				} else {
+					fmt.Fprintln(wout, extensionCmd)
+					fmt.Fprintln(wout, strings.Repeat("-", len(extensionCmd)))
+					fmt.Fprintf(wout, "%s Linkerd extension command %s exists\n", healthcheck.WarnStatus, extensionCmd)
+					fmt.Fprintln(wout)
+					fmt.Fprintf(wout, "Status check results are %s\n", healthcheck.WarnStatus)
```

Comment:
For warning-only failures, the usual `linkerd check` returns OK for the overall success message ("Status check results are √"). So we might do the same here.